packages/ra-core/src/core/CoreAdminRouter.tsx

```diff
@@ -115,111 +119,150 @@ const CoreAdminRouter = (props: AdminRouterProps) => {
-                                )}
+                                {/*
+                                    Render the resources and custom routes that were outside the child function. We use Children.map as it will automatically ignore
+                                    the child function if there is one.
+                                */}
+                                {Children.map(props.children, child => child)}
+                                {/* Render the resources and custom routes that were returned by the child function */}
+                                {childrenToRender}
                             </RoutesWithLayout>
-                        )
-                    }
-                />
-            </Switch>
-        </div>
+                        </Layout>
+                    </div>
+                }
+            />
+        </Routes>
     );
 };
 
 CoreAdminRouter.defaultProps = {
     customRoutes: [],
 };
 
-export default CoreAdminRouter;
+export interface AdminRouterProps extends CoreLayoutProps {
+    layout: LayoutComponent;
+    catchAll: CatchAllComponent;
+    children?: AdminChildren;
+    loading: LoadingComponent;
+    ready?: ComponentType;
+}
+
+const createResourcesFromChildren = (children: React.ReactNode) => {
+    const resources: ResourceDefinition[] = [];
+
+    Children.forEach(children, element => {
+        if (!React.isValidElement(element)) {
+            // Ignore non-elements. This allows people to more easily inline
+            // conditionals in their route config.
+            return;
+        }
+
+        if (element.type === React.Fragment) {
+            // Transparently support React.Fragment and its children.
+            resources.push.apply(
+                resources,
+                createResourcesFromChildren(element.props.children)
+            );
+            return;
+        }
+
+        if (element.type === CustomRoutes) {
+            return;
+        }
+
+        if (element.type !== Resource) {
+            throw new Error(
+                `[${
+                    typeof element.type === 'string'
+                        ? element.type
+                        : element.type.name
+                }] is not a <Route> component. All component children of <Routes> must be a <Route> or <React.Fragment>`
+            );
+        }
+
+        const resourceElement = element as React.ReactElement<ResourceProps>;
+
+        const resource: ResourceDefinition = {
+            name: resourceElement.props.name,
+            options: resourceElement.props.options,
+            hasList: !!resourceElement.props.list,
+            hasEdit: !!resourceElement.props.edit,
+            hasShow: !!resourceElement.props.show,
+            hasCreate: !!resourceElement.props.create,
+            icon: resourceElement.props.icon,
+        };
+
+        resources.push(resource);
+    });
+
+    return resources;
+};
+
+const renderCustomRoutes = (children: ReactNode) => {
```

Comment:
I think you should rename this finction to show that youll only render custom routes with no layout.